administrator/components/com_associations/views/association/tmpl/edit.php

```diff
@@ -33,7 +59,116 @@ function triggerSave() {
 			$('#left-panel').toggle();
 			$('#right-panel').toggleClass('full-width');
 		});
+
+		selectedLang = $('#jform_itemlanguage').val();
+
+		if (selectedLang == '')
+		{
+			document.getElementById('target-association').src = '';
+		}
+
+		//Save button action, replacing the fault Joomla.submitbutton() with custom function
+		$('.btn-success').attr('onclick', function(){
+			var frame = $(this).attr('onclick');
+
+			frame = frame.substring(frame.lastIndexOf('(')+1,frame.lastIndexOf(')'));
+			return 'return triggerSave(' + frame + ')';
+		});
+
+		$('#reference-association').load(function (){
+			//Disable reference fields, commented out just to keep the code for ACL
+			$('#reference-association').each(function () {
+				//$(this).contents().find('.chzn-single').css('background', 'transparent');
+				//$(this).contents().find('.chzn-single').css('background-color', '#eee');
+				//$(this).contents().find('.controls').css( 'pointer-events', 'none' );
+				//$(this).contents().find('input').attr('disabled', 'disabled');
+				$(this).contents().find('#jform_language_chzn').css( 'pointer-events', 'none' );
+				$(this).contents().find('#jform_language_chzn').find('.chzn-single').css('background', 'transparent');
+				$(this).contents().find('#jform_language_chzn').find('.chzn-single').css('background-color', '#eee');
+				$('#toolbar-copy').children().first().attr('onclick', 'return copyRefToTarget()');
+
+				$(this).contents().find('#associations .controls').css( 'pointer-events', 'auto' );
+			});
+		});
+
+		$('#target-association').load(function () {
+			target = $(this);
+
+			//Hide associations tab	
+			target.contents().find('a[href=#associations]').parent().hide();
+
+			langAssociation = '" . str_replace('-', '_', $this->referenceLanguage) . "';
+			langID = " . $this->referenceID . ";
+			target.contents().find('#jform_associations_' + langAssociation + '_id').val(langID);
+
+			split = selectedLang.split('|');
+
+			if (typeof split[1] !== 'undefined') {
+				selectedLang = split[0];
+
+				//Adding checkin IDs to an hidden input
+				if (split[1] != '0') {
+					if (!$('#target-id').val()) {
+						$('#target-id').val(split[1]);
+					}
+					else {
+						$('#target-id').val($('#target-id').val()+','+split[1]);
+					}
+				}
+
+				//Auto-picking the selected language on the switcher
+				target.contents().find('#jform_language option[value='+selectedLang+']').attr('selected','selected');
+				target.contents().find('#jform_language').chosen();
+				target.contents().find('#jform_language').trigger('liszt:updated');
+				target.contents().find('#jform_language').parent().find('div:gt(2)').remove();
```

Comment:
@andrepereiradasilva look at these lines, this must be wrong because I had to hardcode it to perform the behaviour I needed:
Line 120 - I find the '#jform_language' value
Line 121 - Don't know why, it works
Line 122 - Trigger update
Line 123 - I remove an extra language field that was added with .chosen(), if I remove line 121 all these lines are useless